### PR TITLE
`CSVWithNames` with `input_format_with_names_use_header=0`

### LINKED ISSUE #22406: Disabling `input_format_with_names_use_header` discards all the input
**Describe the bug**
When you set the setting `input_format_with_names_use_header` to `0`, `INSERT` does not actually insert rows.

**How to reproduce**
* Which ClickHouse server version to use: git master `a21fe0b492cbc8628126a3cd493b83e82abf9978`, checked out on 2021-03-31
* Which interface to use: the native command-line client
* Non-default settings, if any: `input_format_with_names_use_header` = `0`
* `CREATE TABLE` statements for all tables involved:
```sql
CREATE TABLE csvtest
(
    `t` String
)
ENGINE = MergeTree
ORDER BY t
```
* Sample data for all these tables
`test.csv`:
```csv
t
testdata1
testdata2
```
* Queries to run that lead to unexpected result:
```
% clickhouse-client --input_format_with_names_use_header 0 --query "INSERT INTO csvtest FORMAT CSVWithNames" < test.csv
```

**Expected behavior**
Two rows `('testdata1'), ('testdata2')` will be inserted to the table `csvtest`.

**Error message and/or stacktrace**
Nothing. It silently discards the input and inserts nothing to the table.

**Additional context**
It works as expected if you set `input_format_with_names_use_header` to `1`.

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed the behavior when disabling `input_format_with_names_use_header ` setting discards all the input with CSVWithNames format. This fixes #22406

### CODE REVIEW

**Comment by nikitamikhaylov** [comment]
@Mergifyio rebase

**Review by tavplubix — APPROVED** [review]

File: tests/queries/0_stateless/01818_input_format_with_names_use_header.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+${CLICKHOUSE_CLIENT_BINARY} -q "DROP TABLE IF EXISTS 01818_with_names;"
```
Comment (tavplubix) [inline tests/queries/0_stateless/01818_input_format_with_names_use_header.sh:7]: Did you mean `${CLICKHOUSE_CLIENT}`?

Comment (nikitamikhaylov) [reply tests/queries/0_stateless/01818_input_format_with_names_use_header.sh]: Yes, I copy-pasted this from another test, sorry.

**Comment by nikitamikhaylov** [comment]
I think everything should be Ok.

**Pull request merged**